packages/helper-compiler/src/index.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import { traverse } from "@webassemblyjs/ast";
+import { flatten } from "@webassemblyjs/helper-flatten-ast";
+
+import { Module } from "./module";
+
+export function toIR(ast: Program): IR {
+  const program = {};
+  const funcTable = [];
+
+  // flatten the ast first
+  // TODO(sven): do it in a single pass?
+  flatten(ast);
```

Comment:
The compability with wast might be broken here anyway, we can remove the AST flatten pass.